app.py

```diff
@@ -297,6 +326,32 @@ def parse():
     # reason.
     lang = body.get('lang', requested_lang())
 
+# begin
+    ad_name = body['adventure_name']
+    cur_level = int(body['level'])
+    ad_index = ad_name2index[(cur_level,ad_name)]
+
+    ad_index_db = DATABASE.get_ad(current_user () ['username'])
+    level_db = DATABASE.get_level(current_user () ['username'])
+
+    level_def = load_yaml_rt(f'coursedata/level-defaults/{lang}.yaml')
```

Comment:
level_default? also what exactly does this mean? Why do we need to grab the defaults here, we already have them to render?

Reply:
yes, I just fetch the commands infomation from the level_default so that I can match the input content with the default command. After the user try all the commands, they can unlock the next level.

Reply:
I don't understand this still, what are we grabbing here (maybe add it as a comment in the code)?